packages/qwik/src/core/tests/use-on.spec.tsx

```diff
@@ -647,4 +649,76 @@ describe.each([
     await trigger(document.body, 'div', 'click');
     await expect(document.querySelector('div')).toMatchDOM(<div>1</div>);
   });
+
+  it('issue 7230, when multiple useOnWindow are used in a component that is not rendered, it should add multiple script nodes', async () => {
+    const BreakpointProvider = component$(() => {
+      useOnDocument(
+        'click',
+        $(() => {})
+      );
+
+      useOnWindow(
+        'resize',
+        $(() => {})
+      );
+
+      useVisibleTask$(() => {});
+
+      return <Slot />;
+    });
+
+    const LayoutTest = component$(() => {
+      return (
+        <BreakpointProvider>
+          <div>test</div>
+        </BreakpointProvider>
+      );
+    });
+    const { vNode } = await render(<LayoutTest />, { debug });
+    expect(vNode).toMatchVDOM(
+      <Component ssr-required>
+        <Component ssr-required>
+          <Component ssr-required>
+            <Component ssr-required>
+              <div>test</div>
+            </Component>
+            <script type="placeholder" hidden></script>
+            <script type="placeholder" hidden></script>
+            <script type="placeholder" hidden></script>
+          </Component>
+        </Component>
+      </Component>
+    );
+  });
+  it('issue 7230, when useOnDocument is used in a component that is not rendered, it should add a script node', async () => {
```

Comment:
please rename it to:
`#7230 - when useOnDocument is used in a component that is not rendered, it should add a script node`